### PR TITLE
feat: permission-permit

### PR DESCRIPTION
**Pull request by TheAustrian1998**
- add code for permissionPermit()
- add code for _encode()
- tests

### CODE REVIEW

File: test/unit/DCAStrategies/dca-strategies-permissions-handler.spec.ts [hunk 1]
```diff
@@ -530,4 +530,179 @@ contract('DCAStrategiesPermissionsHandler', () => {
...
+      permissions: [Permission.INCREASE],
+    });
+
+    permitTest({
+      when: `setting two permissions`,
+      permissions: [Permission.REDUCE, Permission.TERMINATE],
+    });
+
+    permitTest({
+      when: `setting three permissions`,
```
Comment (nchamo) [inline test/unit/DCAStrategies/dca-strategies-permissions-handler.spec.ts:585]: Can you add a test that sets four permissions? Just to cover all cases hehe

**Review by nchamo — APPROVED** [review]
Looks good!

File: test/unit/DCAStrategies/dca-strategies-permissions-handler.spec.ts (hunk 1 above)
Comment (TheAustrian1998) [reply test/unit/DCAStrategies/dca-strategies-permissions-handler.spec.ts:585]: Done sir

Comment (nchamo) [reply test/unit/DCAStrategies/dca-strategies-permissions-handler.spec.ts:585]: Merge away, the PR is already approved 😄

**Pull request merged**